arches/app/etl_modules/base_import_module.py

```diff
@@ -367,27 +389,76 @@ def start(self, request):
         return {"success": result["started"], "data": result}
 
     def write(self, request):
-        self.loadid = request.POST.get("load_id")
         self.temp_dir = os.path.join(settings.UPLOADED_FILES_DIR, "tmp", self.loadid)
         self.file_details = request.POST.get("load_details", None)
+        multiprocessing = request.POST.get("multiprocessing", False)
         result = {}
         if self.file_details:
             details = json.loads(self.file_details)
             files = details["result"]["summary"]["files"]
             summary = details["result"]["summary"]
-            use_celery_file_size_threshold_in_MB = 0.1
+            use_celery_file_size_threshold = self.config.get(
+                "celeryByteSizeLimit", 100000
+            )
+
             if (
-                summary["cumulative_files_size"] / 1000000
-                > use_celery_file_size_threshold_in_MB
+                self.mode != "cli"
+                and summary["cumulative_files_size"] > use_celery_file_size_threshold
             ):
                 response = self.run_load_task_async(request, self.loadid)
             else:
                 response = self.run_load_task(
-                    self.userid, files, summary, result, self.temp_dir, self.loadid
+                    self.userid,
+                    files,
+                    summary,
+                    result,
+                    self.temp_dir,
+                    self.loadid,
+                    multiprocessing,
                 )
 
             return response
 
+    def cli(self, source):
+        def return_with_error(error):
+            return {
+                "success": False,
+                "data": {"title": _("Error"), "message": error},
+            }
+
+        read = {"success": False, "message": ""}
+        written = {"success": False, "message": ""}
+
+        initiated = self.start(self.request)
+
+        if initiated["success"]:
+            try:
+                read = self.read(source=source)
+            except Exception as e:
+                return return_with_error(
+                    _("Unexpected Error during reading the files: {}").format(e)
+                )
+        else:
+            return return_with_error(initiated["message"])
+
+        if read["success"]:
+            try:
+                self.request.POST.__setitem__(
+                    "load_details", json.dumps({"result": read["data"]})
+                )
+                written = self.write(self.request)
+            except Exception as e:
+                return return_with_error(
+                    _("Unexpected Error during saving the tiels: {}").format(e)
```

Comment:
Looks like there is a typo in the word "files". Because there is a bit more going on with this step than just saving, I'd change this to `Unexpected error while processing files:`.